herbert_core/utilities/classes/@unique_objects_container/unique_objects_container.m

```diff
@@ -776,6 +822,56 @@ function list(self,field)
                 obj.check_combo_arg(true,true);
             end
         end
+        
+        function out = concatenate(objs, type)
+            %CONCATENATE takes the unique_object and idx (index) arrays from
+            % an array of one or more unique_object_containers and concatenates
+            % separately the unique objects and the indices to single outputs
+            % suitable for use with object_lookup
+            %
+            % Input
+            % -----
+            % - objs - one cell or array of one or more unique_object_containers
+            % - type - '{}' if objs is a cell; anything else (but by
+            %          convention '()') if objs is an array
+            %
+            % Outputs
+            % -------
+            % - out - single unique_objects_container combining the contents of
+            %         the elements of the input array objs
+        
+            if isempty(objs)
+                error('HERBERT:unique_objects_container:invalid_input', ...
+                      'at least one object must be supplied to concatenate');
+            end
+
+            concat_cells = strcmp(type,'{}');
+
+            if numel(objs)==1
+                if concat_cells
+                    out = objs{1};
+                else
+                    out = objs;
+                end
+            else
+                if concat_cells
+                    out = objs{1};
+                    for ii=2:numel(objs)
+                        for jj=1:objs{ii}.n_runs
+                            out = out.add( objs{ii}.get(jj) );
+                        end
+                    end
+                else
+                    out = objs(1);
+                    for ii=2:numel(objs)
+                        for jj=1:objs(ii).n_runs
+                            out = out.add( objs(ii).get(jj) );
+                        end
+                    end
+                end
+            end
```

Comment:
Not sure you need the check for scalar objs. The code will work either way and remove a layer of nesting. Though I wish there was a simple function to modify bracket types rather than `if concat_cells; getter = @(x, y) subsref(x, ['{' num2str(y) '}']); ...`
```suggestion
            if concat_cells
                out = objs{1};
                for ii=2:numel(objs)
                    for jj=1:objs{ii}.n_runs
                        out = out.add( objs{ii}.get(jj) );
                    end
                end
            else
                out = objs(1);
                for ii=2:numel(objs)
                    for jj=1:objs(ii).n_runs
                        out = out.add( objs(ii).get(jj) );
                    end
                end
            end
```

Reply:
we aborted a conversation on distinguishing between an array of unique_object_containers and a single unique_objects_container, and will resume at an appropriate time later